modules/standard/IO.chpl

```diff
@@ -4022,6 +4040,83 @@ proc channel.writebits(v:integral, nbits:integral):bool throws {
   return try this.write(new ioBits(v:uint(64), nbits:int(8)));
 }
 
+proc channel.writeBinary(arg:numeric, param endian:ioendian = ioendian.native) throws {
+  var kind: iokind;
+  var e:syserr = ENOERR;
+
+  select (endian) {
+    when ioendian.native {
+      e = try _write_binary_internal(_channel_internal, iokind.native, arg);      
```

Comment:
It might be nice to update `_write_binary_internal` to take an `ioendian` instead or add a cast between the two types - that way you wouldn't need a select statement, you could just pass the argument through.

Reply:
IMO even if you don't add that, it'd be better to have a select statement computing the iokind, and then just one line after that to call write_binary_internal

Reply:
I wasn't sure of the scope of this PR, I didn't want to touch too many methods. But modifying `_write_binary_internal` seems reasonable. 

As for Mike's comment, I mirrored what is done in `_write_one_internal`, I just assumed there was a good reason for doing it this way.

Reply:
I'm not sure if it will work to compute the param value in a select statement. In the past it wouldn't have, but now it might, with split init. If not, you would have to move it to its own function. Anyway it's not a big deal if the alternatives to what you have are challenging to figure out.

Reply:
I looked into changing `_write_binary_internal` to take an `ioendian` instead, but this change would propagate all the way to `channel.init`. I think it makes more sense to make this change when we remove `iokind` altogether.  What is involved with adding a cast between the two types?

Reply:
You can literally add a cast but I don't think that really matters right now. Here is an example program converting from one enum to another at compile time (with params).

``` chapel
enum enum1 {
  a, b, c
}

enum enum2 {
  a, b, c
}

proc convert(param x: enum1) param : enum2 {
  if x == enum1.a {
    return enum2.a;
  } else if x == enum1.b {
    return enum2.b;
  } else if x == enum1.c {
    return enum2.c;
  }
}

param p = convert(enum1.a);
writeln(p);
```